packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -959,7 +934,124 @@ where
+                r#"
+                -- First we delete the _latest_ entity from the entities table.
+                WITH to_move_to_historic AS (
+                    DELETE FROM entities
+                    WHERE entity_id = $1
+                    RETURNING
+                        entity_id, version,
+                        entity_type_version_id,
+                        properties,
+                        left_order, right_order,
+                        left_entity_id, right_entity_id,
+                        owned_by_id, created_by_id, updated_by_id
+                ),
+                inserted_in_historic AS (
+                    -- We immediately put this deleted entity into the historic table.
+                    -- As this should be done in a transaction, we should be safe that this move
+                    -- doesn't produce invalid state.
+                    INSERT INTO entity_histories(
+                        entity_id, version,
+                        entity_type_version_id,
+                        properties,
+                        left_order, right_order,
+                        left_entity_id, right_entity_id,
+                        owned_by_id, created_by_id, updated_by_id, 
+                        archived
+                    )
+                    SELECT
+                        entity_id, version,
+                        entity_type_version_id,
+                        properties,
+                        left_order, right_order,
+                        left_entity_id, right_entity_id,
+                        owned_by_id, created_by_id, updated_by_id,
+                        $2::boolean
+                    FROM to_move_to_historic
+                    -- For metadata purposes, we return the entire entity
+                    RETURNING *
+                )
+                SELECT
+                    entity_id, inserted_in_historic.version,
+                    base_uri, type_ids.version,
+                    properties, -- Unused
+                    left_order, right_order,
+                    left_entity_id, right_entity_id,
+                    owned_by_id, created_by_id, updated_by_id
+                FROM inserted_in_historic
+                INNER JOIN type_ids ON inserted_in_historic.entity_type_version_id = type_ids.version_id;
+                "#,
+                &[
+                    &entity_id,
+                    &archived,
+                ],
+            )
+            .await
+            .into_report()
+            .change_context(InsertionError)?;
+
+        let link_metadata = match (
+            historic_entity.get(5),
+            historic_entity.get(6),
+            historic_entity.get(7),
+            historic_entity.get(8),
+        ) {
+            (left_order, right_order, Some(left_entity_id), Some(right_entity_id)) => Some(
+                LinkEntityMetadata::new(left_entity_id, right_entity_id, left_order, right_order),
+            ),
+            _ => None,
```

Comment:
This would make sense to do. But i want to point out that we are still missing a check constraint for entities on left and right order. So this could happen in the database without that check. Do you want me to add a check as a drive-by for the orders to only be allowed if the entity_ids are set?

Reply:
hm, I guess it might not hurt to do so. I assume you're saying

> `left_order` is some _only if_ `left_entity_id` is some and `right_entity_id` is some

and 

> `right_order` is some _only if_ `left_entity_id` is some and `right_entity_id` is some

Reply:
Exactly

Reply:
Added check constraint and match case in 5cdd9c9cf290267e1a6933422168acce60fd9783